src/index.jst

```diff
@@ -68,29 +68,16 @@ module.exports = function equal(a, b) {
     if (a.valueOf !== Object.prototype.valueOf) return a.valueOf() === b.valueOf();
     if (a.toString !== Object.prototype.toString) return a.toString() === b.toString();
 
-    keys = Object.keys(a);
-    length = keys.length;
-    if (length !== Object.keys(b).length) return false;
-
-    for (i = length; i-- !== 0;)
-      if (!Object.prototype.hasOwnProperty.call(b, keys[i])) return false;
+    if (Object.keys(a).length !== Object.keys(b).length) return false;
 
-    for (i = length; i-- !== 0;) {
-      key = keys[i];
+    for(key in a){
```

Comment:
Perf improvement here caused omitting the object keys iteration and switching to `for ... in` loop which is faster.

Plus here is fixed false-negative result for situations when key is not own for both of objects.

Reply:
Would be good to have a test case for that.

Same comment here - checking keys in a separate loop is / can be faster for deeper objects?

Also for ES6 version perhaps it’s worth using Reflect.ownKeys to include symbols? (It would simplify the check)

And maybe the equivalent Reflect method to check the presence / value of the key in the second object.

Reply:
Hm >_>
About test cases, thought it was this one: https://codesandbox.io/s/ecstatic-hamilton-sbbew  
But just realized that `Object.keys` returns already own properties, while `for ... in` iterates over inherites props too. Have to go investigate thus.

About `Reflect` usage - haven't thought, btw. Have to investigate too.

Reply:
@epoberezkin About `Reflect`: 
It is ~10 times slower than `Object.keys` (on node 12) mainly because `Object.getOwnPropertySymbols` part.
I suppose it's a bad idea, at least for now.

Reply:
Closing this conversation due to final implementation almost the same with original.